python/docker/README.md

```diff
@@ -0,0 +1,239 @@
+# Docker Build System for OPM Python Wheels
+
+This directory contains the Docker-based build system for generating PyPI-compatible wheels for the OPM Python bindings.
+
+## Directory Structure
+
+```
+docker/
+├── Dockerfile                 # Main Docker build configuration
+├── run-docker-build.sh        # User-facing helper script for building wheels
+├── python_versions.json       # Python version definitions
+├── README.md                  # This file (documentation)
+├── scripts/                   # Internal build scripts
+│   ├── generate-pypi-package.sh    # Wheel generation script
+│   ├── setup-docker-image.sh       # Docker environment setup
+│   ├── build-boost.sh             # Boost dependency build
+│   ├── build-dune.sh              # DUNE dependency build
+│   ├── build-deps.sh              # Additional dependencies
+│   └── read_python_config.py      # Python version config reader
+├── test_wheels/               # Wheel testing framework
+└── [output directories]       # Generated by --output parameter (user-specified names)
+```
+
+## Building Wheels
+
+Use the provided helper script for simplified building with automatic logging:
+
+```bash
+# Static build with wheel extraction
+./docker/run-docker-build.sh static --output wheelhouse
+# Creates: docker/wheelhouse/ with wheels
+
+# Shared libraries build
+./docker/run-docker-build.sh shared --libtype shared --output wheelhouse-shared
+# Creates: docker/wheelhouse-shared/ with wheels
+
+# Build specific Python versions
+./docker/run-docker-build.sh py312 --python-versions "3.12" --output test-wheels
+# Creates: docker/test-wheels/ with wheels
+
+# Debug build with detailed output (no wheel extraction)
+./docker/run-docker-build.sh debug --libtype shared --no-buildx --jobs 8
+
+# Absolute path output (advanced usage)
+./docker/run-docker-build.sh release --output /tmp/release-wheels
+# Creates: /tmp/release-wheels/ with wheels
+
+# Show all options
+./docker/run-docker-build.sh --help
+
+# Test a PR from opm-simulators
+./docker/run-docker-build.sh pr-test --version-simulators "pull/1234/head" --output pr-wheels
+
+# Test multiple PRs together
+./docker/run-docker-build.sh multi-pr \
+    --version-common "pull/567/head" \
```

Comment:
please change head to merge, we do not want to encourage building PR branches directly.